java/src/shared/facades/BuildingFacade.java

```diff
@@ -0,0 +1,176 @@
+package shared.facades;
+
+import shared.locations.EdgeLocation;
+import shared.locations.VertexLocation;
+
+/**
+ * Created by elijahgk on 9/16/2016.
+ * This facade creates an interface to communicate with the underlying building sub-model.
+ */
+public class BuildingFacade {
+
+    /**
+     * Builds a road for the current player at the specified location.
+     * @pre canBuildRoad returns true for the current player and given location.
+     * @post Road is built on game map at specified location.  Current player has one fewer unused roads.
+     * @param buildLocation  The edge on the map where the road is to be built.
+     * @param isFree  Whether the current player gets to build this for free.
+     */
+    public void buildRoad(EdgeLocation buildLocation, boolean isFree){}
```

Comment:
Needs to be build by a specific `Player`, as noted in the Slack chat.
Ditto for all the ones below.